lib/ace/mode/rst_highlight_rules.js

```diff
@@ -37,130 +37,151 @@ var TextHighlightRules = require("./text_highlight_rules").TextHighlightRules;
-       * ACE tokenizer works line per line, it is not possible to have tokens
-       * that spread on multiple lines. */
+       * ACE tokenizer does not allow backtracking, the underlined text of the
+       * heading cannot be highlighted. */
       {
-        token : "markup.heading",
+        token : tokens.title,
         regex : "(^)([\\=\\-`:\\.'\"~\\^_\\*\\+#])(\\2{2,}\\s*$)"
       },
       /* Generic directive syntax (e.g. .. code-block:: c)
        * All of the directive body is highlighted as a code block. */
       {
-        token : ["text", "keyword.operator", "markup.raw"],
-        regex : "(^\\s*)(\\.\\. [^:]+::)(.*$)",
+        token : [tokens.directive, tokens.literal],
+        regex : "(^\\s*\\.\\. [^:]+::)(.*$)",
         next  : "codeblock"
       },
       {
-        token : "keyword.operator",
+        token : tokens.directive,
         regex : "::$",
         next  : "codeblock"
       },
       /* Link/anchor definitions */
       {
-        token : "markup.link",
-        regex : "^\\.\\. _[^:]+:.*$"
+        token : [tokens.entity, tokens.link],
+        regex : "(^\\.\\. _[^:]+:)(.*$)"
       },
       {
-        token : "markup.link",
-        regex : "^__ https?://.*$"
+        token : [tokens.entity, tokens.link],
+        regex : "(^__ )(https?://.*$)"
       },
       /* Footnote definition */
       {
-        token : "markup.list",
+        token : tokens.entity,
         regex : "^\\.\\. \\[[^\\]]+\\] "
       },
       /* Comment block start */
       {
-        token : "comment",
+        token : tokens.comment,
         regex : "^\\.\\. .*$",
         next  : "comment"
       },
       /* List items */
       {
-        token : "markup.list",
-        regex : "^\\s*[\\*\\+-] "
-      },
-      {
-        token : "markup.list",
-        regex : "^\\s*\\d+\\. "
-      },
-      {
-        token : "markup.list",
-        regex : "^\\s*#\\. "
+        token : tokens.list,
+        regex : "^\\s*(?:[\\*\\+-]|\\d+\\.|#\\.) "
       },
       /* "Simple" tables */
       {
-        token : "markup.table",
+        token : tokens.table,
         regex : "^={2,}(?: +={2,})+$"
       },
       /* "Grid" tables */
       {
-        token : "markup.table",
+        token : tokens.table,
         regex : "^\\+-{2,}(?:\\+-{2,})+\\+$"
       },
       {
-        token : "markup.table",
+        token : tokens.table,
         regex : "^\\+={2,}(?:\\+={2,})+\\+$"
       },
       /* Inline markup */
       {
-        token : "markup.raw",
-        regex : "``(?:\\\\`|[^`])+?``"
+        token : tokens.literal,
+        regex : "``",
+        next  : "code"
       },
       {
-        token : "markup.bold",
-        regex : "\\*\\*(?:\\\\\\*|[^\\*])+?\\*\\*"
+        token : tokens.bold,
+        regex : "\\*\\*",
+        next  : "bold"
       },
       {
-        token : "markup.italic",
-        regex : "\\*(?:\\\\\\*|[^\\*])+?\\*"
+        token : tokens.italic,
+        regex : "\\*",
```

Comment:
rst seems to use more complex rules for this.
`*` starts italic span only if it is preceded by space or line start and followed by non whitespace character
empty lines seem to exit italic mode as well.

so maybe we should use something like
```js
{
    token : tokens.italic,
    regex : /( |^)\*(?=[^s*])/,
    next: "italic"
}

"italic": [
    {
        token : tokens.italic,
        regex : /\*(?=\W|$)/,
        next  : "start"
    },
    {
        token : tokens.italic,
        regex : /^\s*$/,
        next  : "start"
    },
    {
        defaultToken: tokens.italic
    }
],